File: graphql/src/main/java/ru/romanow/protocols/graphql/repository/BookRepository.java
package ru.romanow.protocols.graphql.repository;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.repository.query.QueryByExampleExecutor;
import ru.romanow.protocols.graphql.domain.Book;

public interface BookRepository
        extends JpaRepository<Book, Integer>,
                QueryByExampleExecutor<Book> {
}

File: graphql/src/main/java/ru/romanow/protocols/graphql/repository/BookRepository.kt
package ru.romanow.protocols.graphql.repository

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.repository.query.QueryByExampleExecutor
import ru.romanow.protocols.graphql.domain.Book

interface BookRepository : JpaRepository<Book, Int>, QueryByExampleExecutor<Book>